### DISCUSSION TITLE
[A3] extendable only piece-type custom module

### DISCUSSION THREAD

**Discussion by myovchev** [apostrophecms/apostrophe#2903] [Q&A] [answered]
I have this "brilliant" idea of having an `abstract` piece-type module in my project, which does some enhancements to the core `@apostrophecms/piece-type`  without using the `improve` option but I haven't find a good way to do it so far.

What I did is creating a let's call it `custom-piece-type` which extends `@apostrophecms/piece-type`, did my enhancement (fields, filters, columns, etc) and I DIDN'T expose it in the `app.js`, but I made it available to the project via npm module. Now my other pieces are extending the `custom-piece-type` and seems to work as expected.

My question is - is it really so :smile: 
Is it the expected behavior (to work!) and the right way to be done or I'm missing something?

**Comment by boutell** [comment, marked as answer]
Yep! You can extend core types and then have other modules extend your module.
